lib/message/encrypt.js

```diff
@@ -15,5 +15,24 @@ export default function encryptMessage(options) {
     options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
     options.compression = options.compression ? openpgp.enums.compression.zlib : undefined;
 
-    return openpgp.encrypt(options);
+    if (options.detached) {
+        // Create detached signature of message
+        const signature = await options.message.signDetached(options.privateKeys);
+
+        // Encrypt message without signing it (thus no need to change .detached)
+        options.privateKeys = [];
+        const result = await openpgp.encrypt(options);
+
+        // Encrypt signature and add it to the final result
+        options.message = createMessage(signature.packets.write(), options.filename);
+        const encryptedSignature = await openpgp.encrypt(options);
```

Comment:
I think `options.filename` isn't needed here, as it refers to the filename of the data, not of the signature.
Also, perhaps instead of overwriting the original `options` object, you could create a new one here? Something like
```suggestion
        const encryptedSignature = await openpgp.encrypt({
            ...options,
            message: createMessage(signature.packets.write())
        });
```